docs/faq.md

```diff
@@ -202,6 +198,70 @@ jib.from.image = 'custom-base-image'
 
 We currently support adding a custom directory with an **incubating** feature. This feature may change in later versions. If your application needs to use custom files, place them into the `src/main/jib` folder. Files placed here will be added to the filesystem of the container. For example, `src/main/jib/foo/bar` would add `/foo/bar` into the container filesystem.
 
+### I need to add files generated during the build process to a custom directory on the image.
+
+If the current extra directory design doesn't meet your needs (e.g. you need to set up the extra files directory with files generated during the build process), you can use additional goals/tasks to create the extra directory as part of your build.
+
+<details>
+<summary>File copying examples</summary>
+<p>
+
+#### Maven
+
+In Maven, you can use the `maven-resources-plugin` to copy files to your extra directory. For example, if you generate files in `target/generated/files` and want to add them to `/my/files` on the container, you can add the following to your `pom.xml`:
+
+```xml
+<plugins>
+  ...
+  <plugin>
+    <artifact>jib-maven-plugin</artifact>
+    ...
+    <configuration>
+      <extraDirectory>${project.basedir}/target/extra-directory/</extraDirectory>
+    </configuration>
+  </plugin>
+  ...
+  <plugin>
+    <artifact>maven-resources-plugin</artifact>
+    <version>3.1.0</version>
+    <configuration>
+      <outputDirectory>${project.basedir}/target/extra-directory/my/files</outputDirectory>
+      <resources>
+        <resource>
+          <directory>${project.basedir}/target/generated/files</directory>
+        </resource>
+      </resources>
+    </configuration>
+  </plugin>
+  ...
+</plugins>
+```
+
+The `copy-resources` goal will run automatically before compile, so if you are copying files from your build output to the extra directory, you will need to either set the life-cycle phase to `post-compile` or later, or run the goal manually:
+
+```sh
+mvn compile resources:copy-resources jib:build
+```
+
+#### Gradle
+
+The same can be accomplished in Gradle by using a `Copy` task. In your `build.gradle`:
+
+```groovy
+jib.extraDirectory = file('build/extra-directory')
+
+task setupExtraDir(type: Copy) {
```

Comment:
Question: suppose the user wants to make sure this task runs after the normal build is complete. Is that achievable?

Reply:
Probably, using `setupExtraDir.dependsOn`?